pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -500,3 +551,78 @@ func (c *DeploymentConfigController) cleanupOldDeployments(existingDeployments [
 
 	return kutilerrors.NewAggregate(deletionErrors)
 }
+
+// triggerActivated indicates whether we should proceed with new rollout as one of the
+// triggers were activated (config change or image change). The first bool indicates that
+// the triggers are active and second indicates if we should skip the rollout because we
+// are waiting for the trigger to complete update (waiting for image for example).
+func triggerActivated(config *deployapi.DeploymentConfig, latestIsDeployed bool, latestDeployment *v1.ReplicationController) (bool, bool) {
+	if config.Spec.Paused {
+		return false, false
+	}
+	imageTrigger := deployutil.HasImageChangeTrigger(config)
```

Comment:
Is it possible to share code with instantiate?

Reply:
some parts if you talking about canTrigger, ok with follow up? don't want to bloat this PR

Reply:
issue please